iroha_introspect/iroha_introspect_bin/src/main.rs

```diff
@@ -0,0 +1,34 @@
+//! Binary to print all types to json string
+
+#![allow(clippy::print_stdout)]
+
+use std::collections::BTreeMap;
+
+use iroha_data_model::prelude::*;
+use iroha_data_model::query::QueryBox;
+use iroha_introspect::prelude::*;
+
+macro_rules! set_types {
+    ($($t:ty),* $(,)?) => {{
+        let mut out = BTreeMap::new();
+        $(<$t as Introspect>::introspect(&mut out);)*
```

Comment:
Don't we need to use same map for all introspected types for dedup?

Reply:
Yeah, we do. That is why i pass same map to all of types

Reply:
As far as I can see the map created every time from scratch for every type here

Reply:
No it is not. There is a for each type loop at 14 line.